deluge/ui/hostlist.py

```diff
@@ -216,30 +216,39 @@ def on_connect_failed(reason, host_id):
             return defer.succeed(status_offline)
 
         try:
-            ip = gethostbyname(host)
+            # getaddrinfo returns a list of 5-tuples save this to list
+            ips = [addrinfo[4][0] for addrinfo in getaddrinfo(host, None)]
         except gaierror as ex:
             log.warning('Unable to resolve host %s to IP: %s', host, ex.args[1])
             return defer.succeed(status_offline)
+        except IndexError as ex:
+            log.warning('Unable to get ip from host: %s', ex.args[1])
+            return defer.succeed(status_offline)
 
-        host_conn_info = (
-            ip,
-            port,
-            'localclient' if not user and host in LOCALHOST else user,
-        )
-        if client.connected() and host_conn_info == client.connection_info():
-            # Currently connected to host_id daemon.
-            def on_info(info, host_id):
-                log.debug('Client connected, query info: %s', info)
-                return host_id, 'Connected', info
-
-            return client.daemon.info().addCallback(on_info, host_id)
-        else:
-            # Attempt to connect to daemon with host_id details.
-            c = Client()
-            d = c.connect(host, port, skip_authentication=True)
-            d.addCallback(on_connect, c, host_id)
-            d.addErrback(on_connect_failed, host_id)
-            return d
+        host_conn_list = []
+        for host in ips:
+            host_conn_list.append(
+                (
+                    host,
+                    port,
+                    'localclient' if not user and host in LOCALHOST else user,
+                )
+            )
```

Comment:
This can be simplified to a list comprehension 

Also use of `host in ips` is not correct as it overwrites the host check `host in LOCALHOST`. Naming is important to prevent confusion or errors so a better option is `ip` or `host_ip`.

Although it appears that LOCALHOST contains both hostname and ip, although is missing ipv6 `::1`. So the lookup could be replaced with `host_ip in LOCALHOST`